<launch>

    <arg name="visu" default="false"/>
    <arg name="executor" default="ros"/>
    <arg name="bag_repair" default="false"/>
    <arg name="vnet" default="false"/>
    <arg name="auto_start" default="false"/>
    <arg name="alea_file"/>

    <param name="/hidden/aleas"        type="str" textfile="$(arg alea_file)" />

    <include file="hidden-params.launch" />

    <node name="$(anon visu)" pkg="metal" type="onlineTimeline.py" ns="visu"  if="$(arg visu)" />

    <node name="$(anon autoStart)" pkg="metal" type="autoStart.py" ns="autoStart" if="$(arg auto_start)" />

    <node name="$(anon bag)" pkg="rosbag" type="record" args="/hidden/repair -o hidden_repair" if="$(arg bag_repair)" />
    <node name="$(anon bag)" pkg="rosbag" type="record" args="/hidden/stats /hidden/stnvisu /hidden/start /hidden/repair /hidden/mastnUpdate /hidden/communicate  -O $(optenv ROS_LOG_DIR ~/.ros)/stats.bag" />
    
    <node name="$(anon watcher)" pkg="metal" type="watcher.py" required="true" />
    
    <node name="$(anon aleas_injector)" pkg="metal" type="aleaInjector.py" />

    <node name="r1" pkg="metal" type="hidden-ros.py" ns="r1" args="--executor=$(arg executor) --agentName=r1 --waitSignal" >
        <remap from="hidden/repair/in" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/repair/out" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/in" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/out" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
        <remap from="communicate/in" to="/hidden/communicate" unless="$(arg vnet)"/>
        <remap from="communicate/out" to="/hidden/communicate" unless="$(arg vnet)"/>
    </node>

    <node name="r2" pkg="metal" type="hidden-ros.py" ns="r2" args="--executor=$(arg executor) --agentName=r2 --waitSignal" >
        <remap from="hidden/repair/in" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/repair/out" to="/hidden/repair" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/in" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
        <remap from="hidden/mastnUpdate/out" to="/hidden/mastnUpdate" unless="$(arg vnet)"/>
        <remap from="communicate/in" to="/hidden/communicate" unless="$(arg vnet)"/>
        <remap from="communicate/out" to="/hidden/communicate" unless="$(arg vnet)"/>
    </node>

</launch>
